cmd/collector/app/span_processor.go

```diff
@@ -70,7 +79,11 @@ func NewSpanProcessor(
 		sp.processItemFromQueue(value)
 	})
 
-	sp.queue.StartLengthReporting(1*time.Second, sp.metrics.QueueLength)
+	sp.StartStateReporting(1 * time.Second)
```

Comment:
Don't think this function is needed:
```
sp.background(time.Second, sp.updateGauges)
```

Reply:
Not sure I follow. Do you mean that the gofunc + timer (the current code from `sp.background()`) should be duplicated directly in the other functions (`StartDynQueueSizeUpdater`, `StartStateReporting`)?

Reply:
Sorry, I get it now. Updated.